### DISCUSSION TITLE
Vite / Ruby not rendering Vue Page

### DISCUSSION THREAD

**Discussion by HaloElite** [ElMassimo/vite_ruby#184] [Q&A] [answered]
I'm using Ruby 2.5.9 with Rails 4.2.11.3 at my project and try to get my Vue Frontend connected to it using Vite.

`vite version` is giving me:

```
bin/vite present?: true
vite_ruby: 3.0.7
vite_rails_legacy: 3.0.3
rails: 4.2.11.3
node: v17.4.0
npm: 8.3.1
yarn: 1.22.17
pnpm: 
ruby: ruby 2.5.9p229 (2021-04-05 revision 67939) [x86_64-linux]

installed packages:
papierkram@ /home/timo/project/papierkram
├─┬ @vitejs/plugin-vue@1.9.2
│ └── vite@2.7.12 deduped
├─┬ vite-plugin-ruby@3.0.7
│ └── vite@2.7.12 deduped
└── vite@2.7.12
```

The vite.json looks like this:

```
{
  "all": {
    "sourceCodeDir": "frontend",
    "watchAdditionalPaths": []
  },
  "development": {
    "autoBuild": true,
    "publicOutputDir": "vite-dev",
    "port": 3036,
   "host": "127.0.0.1"
  },
  "test": {
    "autoBuild": true,
    "publicOutputDir": "vite-test"
  }
}
```

When now starting with `bin/vite dev` and visiting the frontend I keep getting the error message:

` http://localhost:3000/vite-dev/@id/vue/dist/vue.esm-bundler.js/server-renderer net::ERR_ABORTED 404 (Not Found)`

**Comment by ElMassimo** [comment]
Hi Timo!

It would be useful if you can share a repo, at a first glance the error doesn't seem to be related with Vite Ruby.

Typically, Vue aps running in the browser should not require `server-renderer`, so perhaps there's a misconfiguration in how Vue is being imported.

Also, have you added [`@vitejs/plugin-vue`](https://github.com/ElMassimo/vite_ruby/blob/main/vite-plugin-ruby/example/vite.config.ts#L14) to your Vite configuration?

**Comment by HaloElite** [comment, marked as answer]
@ElMassimo  I can't post the repo because it's my companie's. But the tip with `@vitejs/plugin-vue` did the trick. It was imported but somehow not properly installed using yarn. So thank you really much. You saved my life!
